ledger/acctupdates.go

```diff
@@ -712,6 +711,19 @@ func (au *accountUpdates) committedUpTo(committedRound basics.Round) (retRound b
 
 	offset = uint64(newBase - au.dbRound)
 
+	// check if this update chunk spans across multiple consensus versions. If so, break it so that each update would tackle only a single
+	// consensus version.
+	if au.versions[1] != au.versions[offset] {
```

Comment:
why `version[1]` and not `[0]` ?

Reply:
The versions array is shifted by one ( just like it's `protos` predecessor), and so the indexing there start with one ( and for the same reason - we address au.versions[offset] and not au.versions[offset-1] )

Reply:
and what is in `au.protos[0]`?

Reply:
keep in mind that offset 0, which aligns with `dbRound` is already stored on disk. When we persist rounds, we are going to start off dbRound+1, which perfectly align with the above.